index.js

```diff
@@ -308,17 +309,48 @@ var self = module.exports = {
 				dst = true;
 			}
 		} else {
-			if (date < startDate || date >= endDate) {
-				dst = true;
+			// Important:: Original code had an issue with the case where
+			// current date was greater than the middle of the year and  
+			// the end of dst was in the next year.
+			// Example:: set current date to 06/15/2022 and test with Australia number
+
+			// In the case where end month is within the same year.
+			if(end_month > start_month) {
+				if (date < startDate || date >= endDate) {
+					dst = true;
+				}
+			} else {
+				// In the case where start month is less than the end month which means it is in the next year
+				if (date > startDate || date <= endDate) {
+					dst = true;
+				}
 			}
+
+		}
+
+		object['dstStartDate'] = `${startDate.toISOString()}`;
+		object['dstEndDate'] = `${endDate.toISOString()}`;
+
+		if ( endDate < startDate ) {
+			if ( date < middleOfYear) {
+
+				object['dstStartDate'] = self.nthWeekdayOfMonth(start_day, start_week, new Date(year-1, start_month)).toISOString();
+			} else {
+				object['dstEndDate'] = self.nthWeekdayOfMonth(end_day, end_week, new Date(year+1, end_month)).toISOString();
+			}
+
 		}
+
 		object['dstnow'] = dst;
 		return dst;
 	},
 	nthWeekdayOfMonth: function (weekday, n, date) {
 		var count = 0;
 		var result = new Date(date.getFullYear(), date.getMonth(), 1);
 		while (true) {
+			if (typeof weekday !== 'number' || isNaN(weekday)) {
```

Comment:
Add a comment before if statement that states the following:

// If statement added to stop infinite loop for certain numbers in middle east

Reply:
comment added, thank you.